spec/spec.md

```diff
@@ -58,6 +66,40 @@ the same type.
 
 `union(...)` is the supertype of all its operand types.
 
+### Values, Locations, and Cells
+
+A *value* is something that inhabits a type. They're either constructed
+implicitly or are produced by terms. An *aggregate value* is a value
+inhabiting a composite type.
+
+*Values* are stored in *locations*. If a location stores an aggregate value,
+it consists of one or more sub-locations. A location not part of any other
+location is called a *cell*.
+
+Every implicitly or explicitly constructed value has a unique *identity*,
+meaning two separate values cannot be the *same*, but they can be *equal*.
```

Comment:
Overloading the term "value" like this (i.e., having it also refer to a run-time *instance* of a value) is a bad idea, in my opinion. A separate term would be better, though I haven't been able to come up with a satisfactory one yet.

A possible candidate could be "value instance". C uses the term ["object"](https://cppreference.com/en/c/language/object.html) for this concept.